pandas/core/generic.py

```diff
@@ -198,6 +199,8 @@
     from pandas.core.indexers.objects import BaseIndexer
     from pandas.core.resample import Resampler
 
+    _NDFrameT = TypeVar("_NDFrameT", bound="NDFrame")
```

Comment:
can you add a comment on why this is necessary?  e.g. i think once Self is available we can swap this out?

Reply:
Yes, this can be removed when we can use `Self`. If we want to link two parameters to be of the same NDFrame (sub-class) `NDFrameT` is enough, but if we have two pairs, we need to use a second TypeVar for the second pair. Otherwise, all four parameters are bound to the same (sub)class.

Reply:
i know, im asking for a comment in the code to that effect

Reply:
Added a comment. I realized that we cannot remove it when Self is available as Self doesn't help `_align_as_utc`.

Reply:
Probably better to move it to `pandas._typing` since this will not be a temporary fix.

Reply:
i think when the time comes we can make _align_as_utc a NDFrame method

Reply:
moved to pandas._typing